// yuk/src/model/ComPatientsModel.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Yuk) {
      root.Yuk = {};
    }
    root.Yuk.ComPatientsModel = factory(root.Yuk.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ComPatientsModel model module.
   * @module model/ComPatientsModel
   * @version 1.1.1
   */

  /**
   * Constructs a new <code>ComPatientsModel</code>.
   * @alias module:model/ComPatientsModel
   * @class
   * @param name
   */
  var exports = function(name) {


    this['name'] = name;



  };

  /**
   * Constructs a <code>ComPatientsModel</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ComPatientsModel} obj Optional instance to populate.
   * @return {module:model/ComPatientsModel} The populated <code>ComPatientsModel</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'Integer');
      }
      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
      if (data.hasOwnProperty('treatcard')) {
        obj['treatcard'] = ApiClient.convertToType(data['treatcard'], 'String');
      }
      if (data.hasOwnProperty('idcard')) {
        obj['idcard'] = ApiClient.convertToType(data['idcard'], 'String');
      }
      if (data.hasOwnProperty('phone')) {
        obj['phone'] = ApiClient.convertToType(data['phone'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {Integer} id
   */
  exports.prototype['id'] = undefined;

  /**
   * @member {String} name
   */
  exports.prototype['name'] = undefined;

  /**
   * @member {String} treatcard
   */
  exports.prototype['treatcard'] = undefined;

  /**
   * @member {String} idcard
   */
  exports.prototype['idcard'] = undefined;

  /**
   * @member {String} phone
   */
  exports.prototype['phone'] = undefined;




  return exports;
}));
